src/preamble.js

```diff
@@ -2401,7 +2403,13 @@ function integrateWasmJS(Module) {
   Module['asmPreload'] = Module['asm'];
 
   // Memory growth integration code
+
+  var defaultReallocBuffer = Module['reallocBuffer'];
+
   Module['reallocBuffer'] = function(size) {
+    if (finalMethod === 'asmjs') {
+      return defaultReallocBuffer(size);
```

Comment:
Also, the name "default" for `defaultReallocBuffer` is a bit counterintuitive. This reads to me as an overriding method instead of a default method. However, would it be possible to have two separate functions `function wasmReallocBuffer` and `function asmjsReallocBuffer`, and depending on which binaryen method succeeds, assign the appropriate version to `Module['reallocBuffer']`? That would be a more straightforward way? The current method looks a bit inconvenient due to its hot-patching style since it introduced two new state variables to do the job. (Fine either way, though I felt the logic here was a bit cryptic to read, but might be just me having a slow Sunday..)